README.md

```diff
@@ -74,26 +74,48 @@ Environment variables
 
 Real Inputs
 ---------------
-To hook the server up to your real inputs you need to create a configuration file that connects to your input source and applies the relevant parsers / converters in the provider pipeline.
+To hook the server up to your real inputs you need to create a configuration file that connects to your input source and applies the relevant parsers / converters in the provider pipeline.it is
 
-Inputs are configured as an array of pipedProviders, each with an id and an array pipeElements. You need to create a pipedProviders entry for each of your inputs.
+Inputs are configured as an array of pipedProviders, each with an id and an array pipeElements. You need to create a pipedProviders entry for each of your inputs. The pipedProvider starts with a name (name of the interface, e.g. NMEA0183 from AIS could be called "AIS"). The first provider is the source of the data ("providers/serialport" for a serial interface such as USB, "providers/filestream" if the data comes from a file, "providers/tcp" if the source is a TCP port etc.). You will see in the example settings that the second provider is often "providers/liner". This is a provider that splits the input into separate lines and passes one line at a time to the next provider. The final provider or group of providers is where the translation or passing to the server occurs. For NMEA0183, this is "providers/nmea0183-signalk", for N2K (NMEA2000) it is "providers/n2kAnalyzer" and then "providers/n2k-signalk".
```

Comment:
What did you mean with `name? There is only id, no name.

Strictly speaking the provider does not _start_ with the id, because the id and pipeElements properties can be in any order.

I think it would be good to talk about `pipeElements` that make up a `PipedProvider`, even if they are loaded from a directory called providers.